### PR TITLE
[stable/3.0] backport base of api 2.0

### PR DESCRIPTION
**Pull request by MaximilianMeister**
but only the endpoints, that need to run on cloud 6

### CODE REVIEW

**Comment by jsuchome** [comment]
LGTM: we need to backport parts of 2.0 API, but I cannot judge if this is enough/100% correct.

**Comment by MaximilianMeister** [comment]
> LGTM: we need to backport parts of 2.0 API, but I cannot judge if this is enough/100% correct.

there is still the repo check api endpoint missing, but will add it later on to master and backport it

File: crowbar_framework/app/controllers/api/errors_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#
```
Comment (rsalevsky) [inline crowbar_framework/app/controllers/api/errors_controller.rb]: I would not backport this. It can create new regressions on cloud6.

**Comment by jsuchome** [comment]
+1

**Comment by rsalevsky** [comment]
+1

**Comment by MaximilianMeister** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;xeN0LnPm&#x2F;95-5-s47p9-implement-api-for-preparing-nodes-for-upgrade-upgrade-step-2"> (5) S47P9 Implement API for preparing nodes for upgrade (upgrade step 2)</a>

**Pull request merged**